src/main/java/org/jabref/gui/groups/GroupDialogViewModel.java

```diff
@@ -107,12 +107,14 @@
     private final PreferencesService preferencesService;
     private final BibDatabaseContext currentDatabase;
     private final AbstractGroup editedGroup;
+    private final boolean isGroup;
 
-    public GroupDialogViewModel(DialogService dialogService, BibDatabaseContext currentDatabase, PreferencesService preferencesService, AbstractGroup editedGroup) {
+    public GroupDialogViewModel(DialogService dialogService, BibDatabaseContext currentDatabase, PreferencesService preferencesService, AbstractGroup editedGroup, boolean isGroup) {
         this.dialogService = dialogService;
         this.preferencesService = preferencesService;
         this.currentDatabase = currentDatabase;
         this.editedGroup = editedGroup;
+        this.isGroup = isGroup;
```

Comment:
Is this really necessary? Can't be determined if this group is a subgroup in checking if the editedGroup has a parent other than the root node?

Reply:
We added another variable to identify if it is a group or a subgroup. editGroup doesn't classify whether each one of them.

Reply:
we are currently converting the boolean to enum as per suggested by @Siedlerchr